### PR TITLE
saucelab intergation

### PR DESCRIPTION
**Pull request by valorkin**
As suggested in saucelab cookbook, I created saucelab configuraion (karma-sauce-launcheer) into karma.conf-ci.js and created like this
```bash
“test:saucelabs”: “./node_modules/@angular/cli/bin/ng test core --watch false --code-coverage true --browsers=ChromeHeadless --karmaConfig=karma.conf-ci.js”
and tried
“test:saucelabs”: “./node_modules/@angular/cli/bin/ng test core --watch false --code-coverage true --browsers=SauceLabs --karmaConfig=karma.conf-ci.js”
is displaying in console,
(node:13613) ExperimentalWarning: The dns.promises API is experimental
(node:13613) ExperimentalWarning: The fs.promises API is experimental
28 07 2020 21:10:31.015:INFO [karma-server]: Karma v5.1.0 server started at http://0.0.0.0:9876/
28 07 2020 21:10:31.018:INFO [launcher]: Launching browsers ChromeHeadless with concurrency unlimited
28 07 2020 21:10:31.023:INFO [launcher]: Starting browser ChromeHeadless
28 07 2020 21:10:31.783:INFO [Chrome Headless 84.0.4147.89 (Mac OS 10.15.6)]: Connected on socket qPk8DC_D4OCkLgElAAAA with id 48747519
Chrome Headless 84.0.4147.89 (Mac OS 10.15.6) ERROR
  You need to include some adapter that implements __karma__.start method!
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fundamental-ngx@0.20.1-rc.33 test:saucelabs: `./node_modules/@angular/cli/bin/ng test core --watch false --code-coverage true --browsers=ChromeHeadless --karmaConfig=karma.conf-ci.js`
```

and in case of browsers as Saucelab

```bash
> ./node_modules/@angular/cli/bin/ng test core --watch false --code-coverage true --browsers=SauceLabs --karmaConfig=karma.conf-ci.js
(node:17124) ExperimentalWarning: The dns.promises API is experimental
(node:17124) ExperimentalWarning: The fs.promises API is experimental
29 07 2020 09:52:30.020:INFO [karma-server]: Karma v5.1.0 server started at http://0.0.0.0:9876/
29 07 2020 09:52:30.022:INFO [launcher]: Launching browsers SauceLabs with concurrency unlimited
29 07 2020 09:52:30.025:INFO [launcher]: Starting browser undefined on SauceLabs
(node:17124) ExperimentalWarning: Readable[Symbol.asyncIterator] is an experimental feature. This feature could change at any time
2020-07-29T04:23:03.624Z ERROR webdriver: Request failed with status 400 due to Error: No browserName or device specified in session request.
```

conf-ci has below browser listed.

```ts
const customLaunchers = {
        'SL_Chrome': {
            base: 'SauceLabs',
            browserName: 'chrome',
            version: 'latest'
        }
```
could you please suggest on this

### CODE REVIEW

File: sauce.conf.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        ChromeHeadlessNoSandbox: {
+            base: "SauceLabs",
+            flags: [
+                "--no-sandbox",
+                // required to run without privileges in Docker
+                "--disable-web-security",
+                "--disable-gpu",
+                "--remote-debugging-port=9222"
+            ]
+        },
```
Comment (elfro) [inline sauce.conf.js:22]: need to delete this item from arr cause ChromeHeadlessNoSandbox is unknown launcher for the SauceLab

File: sauce.conf.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+const { join } = require('path');
+const { constants } = require('karma');
+
+module.exports = () => {
```
Comment (elfro) [inline sauce.conf.js:7]:
```suggestion
module.exports = (config) => {
```

File: sauce.conf.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        // },
+    }
+
+    // If we're running inside Travis we will set the TRAVIS_BUILD_NUMBER and TRAVIS_BUILD_ID as the Sauce Labs build
+    // This will allow us to easily link back from Sauce Labs to Travis
+    if (process.env.TRAVIS_BUILD_NUMBER && process.env.TRAVIS_BUILD_ID) {
+        sauceLabs.build = `TRAVIS #${process.env.TRAVIS_BUILD_NUMBER} (${process.env.TRAVIS_BUILD_ID})`;
+    }
+
+    return {
```
Comment (elfro) [inline sauce.conf.js:66]:
```suggestion
    config.set({
```

File: sauce.conf.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+        specReporter: {
+            maxLogLines: 5,             // limit number of lines logged per test
+            suppressErrorSummary: true, // do not print error summary
+            suppressFailed: false,      // do not print information about failed tests
+            suppressPassed: false,      // do not print information about passed tests
+            suppressSkipped: true,      // do not print information about skipped tests
+            showSpecTiming: false           // test would finish with error when a first fail occurs.
+        }
+    };
```
Comment (elfro) [inline sauce.conf.js:136]:
```suggestion
    });
```

File: package.json [hunk 5]
```diff
@@ -36,6 +36,7 @@
     "release:create": "create-release",
     "start": "ng serve",
     "test": "ng test core --watch false --code-coverage true --browsers=ChromeHeadless",
+    "test:saucelabs": "ng test core --watch false --code-coverage true --browsers=SauceLabs --karmaConfig=sauce.conf.js",
```
Comment (elfro) [inline package.json:39]:
```suggestion
    "test:saucelabs": "ng test core --watch false --code-coverage true --karmaConfig=sauce.conf.js",
```

Comment (sKudum) [reply package.json]: Hi Ellina, 
Thank You for prompt response :-).
When I explored more I found that it is not able to get the spec.ts files to execute at all.
basically, in sauce, all these test specs has to get executed. I tried file patterns and spec reporters but non of them worked. 

All spects of present inside lib/ <core|platform>............*.spec.ts.


I am reverting back the code, It has same approach as you suggested.
 files: [
            //  { pattern: '/\.spec\.ts/', included: true, watched: true }
            {
                pattern: '../libs/core/src/lib/*/*\.spec\.ts'
            },
            {
                pattern: '../libs/platform/src/lib/components/*/*\.spec\.ts'
            }
        ],

**Pull request closed without merging**